### PR TITLE
[Merged by Bors] - chore: deprecate Cardinal.mk'_def

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by vihdzp** [comment]
Pushed some minor golfs.

**Review by vihdzp — APPROVED** [review]

File: Mathlib/SetTheory/Ordinal/Basic.lean [hunk 1]
```diff
@@ -1135,11 +1135,10 @@ theorem lt_ord {c o} : o < ord c ↔ o.card < c :=
...
 @[simp]
-theorem card_ord (c) : (ord c).card = c :=
-  Quotient.inductionOn c fun α => by
-    let ⟨r, _, e⟩ := ord_eq α
-    -- Porting note: cardinal.mk_def is now Cardinal.mk'_def, not sure why
-    simp only [mk'_def, e, card_type]
+theorem card_ord (c) : (ord c).card = c := by
+  refine c.inductionOn fun α ↦ ?_
+  let ⟨r, _, e⟩ := ord_eq α
+  rw [e, card_type]
```
Comment (b-mehta) [inline Mathlib/SetTheory/Ordinal/Basic.lean:1141]:
```suggestion
theorem card_ord (c) : (ord c).card = c := by
  c.inductionOn fun α ↦ let ⟨r, _, e⟩ := ord_eq α; e ▸ card_type r
```

**Review by b-mehta — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/SetTheory/Ordinal/Basic.lean [hunk 2]
```diff
@@ -1135,11 +1135,8 @@ theorem lt_ord {c o} : o < ord c ↔ o.card < c :=
   gc_ord_card.lt_iff_lt
 
 @[simp]
-theorem card_ord (c) : (ord c).card = c :=
-  Quotient.inductionOn c fun α => by
-    let ⟨r, _, e⟩ := ord_eq α
-    -- Porting note: cardinal.mk_def is now Cardinal.mk'_def, not sure why
-    simp only [mk'_def, e, card_type]
+theorem card_ord (c) : (ord c).card = c := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/SetTheory/Ordinal/Basic.lean:1138]:
```suggestion
theorem card_ord (c) : (ord c).card = c :=
```

**Comment by Ruben-VandeVelde** [comment]
bors merge

**Pull request closed without merging**